### PR TITLE
fix(components): Vlist and SortBy fixes

### PR DESCRIPTION
**Pull request by smouillour**
**What is the problem this PR is trying to solve?**
SortBy : Visible only with display mode = 'large'
Vlist : Manage SortParams
**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/List/ListComposition/SortBy/SortBy.component.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function SortBy(props) {
 	const onOrderChange = event =>
 		performChange(event, { ...currentValue, isDescending: !currentValue.isDescending });
 
-	return (
+	return displayMode !== 'table' ? (
```
Comment (frassinier) [inline packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: I hope we can use a constant for that

File: packages/components/src/List/ListComposition/VList/VList.component.js [hunk 2]
```diff
@@ -4,8 +4,19 @@ import { useListContext } from '../context';
...
 function VList(props) {
-	const { displayMode = 'table', collection } = useListContext();
-	return <VirtualizedList collection={collection} type={displayMode.toUpperCase()} {...props} />;
+	const { displayMode = 'table', collection, setSortParams, sortParams } = useListContext();
+	return (
+		<VirtualizedList
+			collection={collection}
+			type={displayMode.toUpperCase()}
+			sortBy={sortParams.sortBy}
+			sortDirection={sortParams.isDescending ? 'DESC' : 'ASC'}
```
Comment (frassinier) [inline packages/components/src/List/ListComposition/VList/VList.component.js]: Export const for sort direction too

File: packages/components/src/List/ListComposition/SortBy/SortBy.component.js (hunk 1 above)
Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: yes we can

Comment (jsomsanith-tlnd) [inline packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: Can you avoid the ternary and use a fast return ?
```javascript
if (displayMode === 'table') { // or a constant as suggested by frassinier
    return null;
}

return ...
```

Comment (frassinier) [inline packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: We can return `null` before all of that stuff by using
```javascript
if (displayMode === DISPLAY_MODE.TABLE) {
  return null;
}

useEffect();  // bla bla bla
```

Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: done

Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: done

Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: done

File: packages/components/src/List/ListComposition/VList/VList.component.js (hunk 2 above)
Comment (smouillour) [reply packages/components/src/List/ListComposition/VList/VList.component.js]: done

File: packages/components/src/List/ListComposition/SortBy/SortBy.component.js [hunk 3]
```diff
@@ -40,6 +41,9 @@ function SortBy(props) {
 	const onOrderChange = event =>
 		performChange(event, { ...currentValue, isDescending: !currentValue.isDescending });
 
+	if (displayMode === DISPLAY_MODE.TABLE) {
```
Comment (frassinier) [inline packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: If you move that to line 12, I will be really happy 👍

Comment (jsomsanith-tlnd) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: Not on line 12 because all hooks must run. But line 20 yes

Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: done

File: packages/components/src/List/ListComposition/VList/VList.component.js [hunk 4]
```diff
@@ -2,10 +2,22 @@ import React from 'react';
 
 import { useListContext } from '../context';
 import VirtualizedList from '../../../VirtualizedList';
+import { SORT } from '../constants';
```
Comment (frassinier) [inline packages/components/src/List/ListComposition/VList/VList.component.js]:
```suggestion
import { DISPLAY_MODE, SORT } from '../constants';
```

File: packages/components/src/List/ListComposition/VList/VList.component.js [hunk 5]
```diff
@@ -2,10 +2,22 @@ import React from 'react';
...
 
 function VList(props) {
-	const { displayMode = 'table', collection } = useListContext();
-	return <VirtualizedList collection={collection} type={displayMode.toUpperCase()} {...props} />;
+	const { displayMode = 'table', collection, setSortParams, sortParams } = useListContext();
```
Comment (frassinier) [inline packages/components/src/List/ListComposition/VList/VList.component.js]:
```suggestion
	const { displayMode = DISPLAY_MODE.TABLE, collection, setSortParams, sortParams } = useListContext();
```

File: packages/components/src/List/ListComposition/VList/VList.component.js (hunk 4 above)
Comment (smouillour) [reply packages/components/src/List/ListComposition/VList/VList.component.js]: done

File: packages/components/src/List/ListComposition/VList/VList.component.js (hunk 5 above)
Comment (smouillour) [reply packages/components/src/List/ListComposition/VList/VList.component.js]: done

File: packages/components/src/List/ListComposition/SortBy/SortBy.component.js [hunk 6]
```diff
@@ -3,11 +3,17 @@ import PropTypes from 'prop-types';
...
 
+import { DISPLAY_MODE } from '../constants';
 import { useListContext } from '../context';
 
 function SortBy(props) {
 	const { id, initialValue, options, onChange, value } = props;
-	const { sortParams, setSortParams, t } = useListContext();
+	const { displayMode, sortParams, setSortParams, t } = useListContext();
+
+	if (displayMode === DISPLAY_MODE.TABLE) {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: You break the rules of hooks.
Move that after the useEffect please

Comment (smouillour) [reply packages/components/src/List/ListComposition/SortBy/SortBy.component.js]: done

**Pull request merged**